include/arch/x86/linker.ld

```diff
@@ -307,15 +307,15 @@ SECTIONS
 #include <linker/kobject.ld>
 
 	__data_ram_end = .;
-
-	/* All unused memory also owned by the kernel for heaps */
-	__kernel_ram_end = PHYS_RAM_ADDR + KB(CONFIG_RAM_SIZE);
+	__kernel_ram_end = .;
 	__kernel_ram_size = __kernel_ram_end - __kernel_ram_start;
 
 	_image_ram_end = .;
 	_image_ram_all = (PHYS_RAM_ADDR + KB(CONFIG_RAM_SIZE)) - _image_ram_start;
 
-	_end = .; /* end of image */
+	MMU_PAGE_ALIGN
+	_end = .; /* end of image, used to begin the newlib heap */
+	_newlib_max_heap = (PHYS_RAM_ADDR + KB(CONFIG_RAM_SIZE)) - _end;
```

Comment:
is this the size of the heap?  If so, the name needs to denote that.

Reply:
it's the maximum size of the heap. by default, the internal pointers tracked by _sbrk() point to _end, _newlib_max_heap is how far out it can ultimately go.